src/arithmetic/map_funcs/map_funcs.c

```diff
@@ -74,6 +74,30 @@ SIValue AR_PROPERTIES(SIValue *argv, int argc, void *private_data) {
 	return SI_NullVal();
 }
 
+// Receives two maps and merges them
+SIValue AR_MERGEMAP(SIValue *argv, int argc, void *private_data) {
+	ASSERT(argc == 2);
+
+	SIValue map0 = argv[0];
+	SIValue map1 = argv[1];
+
+	if ((SI_TYPE(map0) & T_NULL) && (SI_TYPE(map1) & T_NULL)) {
+		return SI_NullVal();
+	} else if (SI_TYPE(map0) & T_NULL) {
+		return map1;
+	} else if (SI_TYPE(map1) & T_NULL) {
+		return map0;
+	} else {
+		uint keyCount0 = Map_KeyCount(map0);
+		SIValue map = Map_Clone(map1);
```

Comment:
I'm using Map_Clone() because when the code was adding values to existing maps I detected errors with this test:
```
127.0.0.1:6379>  graph.query g "CREATE (:L {val:1})-[:E]->(:L {val:2})-[:E]->(:L {val:3})"
1) 1) "Labels added: 1"
   2) "Nodes created: 3"
   3) "Properties set: 3"
   4) "Relationships created: 2"
   5) "Cached execution: 0"
   6) "Query internal execution time: 1.368740 milliseconds"
127.0.0.1:6379>  graph.query g "MATCH (n) RETURN n{.*, .h}"
1) 1) "n{.*, .h}"
2) 1) 1) "{h: 1, h: NULL}"
   2) 1) "{val: 2, val: NULL}"
   3) 1) "{val: 3, h: NULL}"
3) 1) "Cached execution: 0"
   2) "Query internal execution time: 0.391013 milliseconds"
```